magento/magento2
# Magento should create a log entry if an observer does not implement ObserverInterface

### Description (*)
In developer mode, if an observer does not implement ObserverInterface, Magento throws an exception. This is the expected behavior.

What I don't understand is that it does not do anything for other deploy modes (default and production). It silently discards the observer without even putting an entry to logs.

### Fixed Issues (if relevant)

1. magento/magento2#21755: Magento should create a log entry if an observer does not implement ObserverInterface 

### Manual testing scenarios (*)

1. Create an Observer class wich not implements ObserverInterface
2. Check Log sytem.log

### Contribution checklist (*)
